Src/BootCamp.Chapter/BinaryConverter.cs

```diff
@@ -1,19 +1,42 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Text;
+using System.Text.RegularExpressions;
 
 namespace BootCamp.Chapter
 {
     public static class BinaryConverter
     {
         public static long ToInteger(string binary)
         {
-            return 0;
+            if (string.IsNullOrEmpty(binary)) return 0;
+            if (!Regex.IsMatch(binary, @"^[01]+$")) throw new InvalidBinaryNumberException(binary);
+
+            var num = 0;
+            foreach (var bin in binary)
+            {
+                if (!int.TryParse(bin.ToString(), out var value)) throw new ArgumentOutOfRangeException("An unexpected error occurred");
```

Comment:
If exception message is ever going to be just that, please don't give any message. It's noise. And besides, in this case don't use int.TryParse, just use int.Parse and it will fail giving more info than you give right now.